py4j-python/src/py4j/java_gateway.py

```diff
@@ -2110,14 +2120,16 @@ def launch_gateway(
             redirect_stdout=redirect_stdout, redirect_stderr=redirect_stderr,
             daemonize_redirect=daemonize_redirect, java_path=java_path,
             create_new_process_group=create_new_process_group,
-            enable_auth=enable_auth)
+            enable_auth=enable_auth, return_proc=True)
         if enable_auth:
-            _port, _auth_token = _ret
+            _port, _auth_token, proc = _ret
         else:
-            _port, _auth_token = _ret, None
+            _port, proc, _auth_token = _ret + (None, )
         gateway = JavaGateway(
             gateway_parameters=GatewayParameters(port=_port,
                                                  auth_token=_auth_token))
+        # NOTE 3: the Popen object is now available with gateway._proc
+        gateway._proc = proc
```

Comment:
could you add this member to JavaGateway.__init__ initialized to None so that even if the gateway is not launched through launch_gateway, the member is defined?

Reply:
Sounds good. I'm calling the new the `JavaGateway` member `java_process` by analogy with `java_gateway_server`, if that sounds like a good name.